src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -518,6 +525,12 @@ private void validateCvrFileSources() {
             isValid = false;
             Logger.log(Level.SEVERE, "CDF files must be tabulated individually.");
           }
+          if (!source.getFilePath().endsWith(".xml") && !source.getFilePath().endsWith(".json")) {
+            Logger
+                .severe("CDF source files must be .json or .xml! Unexpected file extension for: %s",
+                    source.getFilePath());
+            isValid = false;
+          }
```

Comment:
Why are we checking this again here? Isn't it redundant with lines 209-216?

Reply:
Yeah it does seem redundant.  @HEdingfield is that right?

Reply:
Yeah, Hylton mentions this below.